geode-lucene/src/test/java/org/apache/geode/cache/lucene/RebalanceWithRedundancyWithRegionCreatedBeforeReindexDUnitTest.java

```diff
@@ -0,0 +1,343 @@
+import org.apache.geode.cache.lucene.internal.LuceneIndexFactoryImpl;
+import org.apache.geode.cache.lucene.internal.LuceneServiceImpl;
+import org.apache.geode.cache.partition.PartitionMemberInfo;
+import org.apache.geode.cache.partition.PartitionRebalanceInfo;
+import org.apache.geode.cache.partition.PartitionRegionInfo;
+import org.apache.geode.test.dunit.AsyncInvocation;
+import org.apache.geode.test.dunit.Host;
+import org.apache.geode.test.dunit.SerializableRunnable;
+import org.apache.geode.test.dunit.SerializableRunnableIF;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.junit.categories.DistributedTest;
+
+@Category(DistributedTest.class)
+@RunWith(JUnitParamsRunner.class)
+public class RebalanceWithRedundancyWithRegionCreatedBeforeReindexDUnitTest
+    extends LuceneQueriesAccessorBase {
+
+  protected VM dataStore3;
+  protected VM dataStore4;
+
+  public void postSetUp() throws Exception {
+    super.postSetUp();
+    dataStore3 = Host.getHost(0).getVM(2);
+    dataStore4 = Host.getHost(0).getVM(3);
+  }
+
+  @Before
+  public void setNumBuckets() {
+    NUM_BUCKETS = 113;
+  }
+
+  @Before
+  public void setLuceneReindexFlag() {
+    dataStore1.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = true);
+    dataStore2.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = true);
+    dataStore3.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = true);
+    dataStore4.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = true);
+  }
+
+  @Override
+  protected RegionTestableType[] getListOfRegionTestTypes() {
+    return new RegionTestableType[] {RegionTestableType.PARTITION,
+        RegionTestableType.PARTITION_REDUNDANT};
+  }
+
+  @After
+  public void clearLuceneReindexFlag() {
+    dataStore1.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = false);
+    dataStore2.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = false);
+    dataStore3.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = false);
+    dataStore4.invoke(() -> LuceneServiceImpl.LUCENE_REINDEX = false);
+  }
+
+  protected SerializableRunnable createIndex = new SerializableRunnable("createIndex") {
+    public void run() {
+      LuceneService luceneService = LuceneServiceProvider.get(getCache());
+      ((LuceneIndexFactoryImpl) luceneService.createIndexFactory()).addField("text")
+          .create(INDEX_NAME, REGION_NAME, LuceneServiceImpl.LUCENE_REINDEX);
+    }
+  };
+
+  protected SerializableRunnable rebalance = new SerializableRunnable("rebalance") {
+    public void run() {
+      Cache cache = getCache();
+      cache.getRegion(REGION_NAME);
+
+      ResourceManager resMan = cache.getResourceManager();
+      RebalanceFactory factory = resMan.createRebalanceFactory();
+      System.out.println("Starting rebalance");
+      RebalanceResults rebalanceResults = null;
+      RebalanceOperation rebalanceOp = factory.start();
+      try {
+        rebalanceResults = rebalanceOp.getResults();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```

Comment:
Should we fail the test if re-balance encountered an Exception ??

Reply:
maybe re-throw exception so that it caught in the main tests checkException() method.

Reply:
Good catch, thanks!